homeassistant/components/airvisual_pro/sensor.py

```diff
@@ -0,0 +1,165 @@
+    ),
+    SensorEntityDescription(
+        key=SENSOR_KIND_PM_1_0,
+        name="PM 1.0",
+        device_class=SensorDeviceClass.PM10,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=SENSOR_KIND_PM_2_5,
+        name="PM 2.5",
+        device_class=SensorDeviceClass.PM25,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=SENSOR_KIND_TEMPERATURE,
+        name="Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        key=SENSOR_KIND_VOC,
+        name="VOC",
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+@callback
+def async_get_aqi_locale(settings: dict[str, Any]) -> str:
+    """Return the correct AQI locale based on settings data."""
+    if settings["is_aqi_usa"]:
+        return "aqi_us"
+    return "aqi_cn"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up AirVisual sensors based on a config entry."""
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        AirVisualProSensor(coordinator, description)
+        for description in SENSOR_DESCRIPTIONS
+    )
+
+
+class AirVisualProSensor(AirVisualProEntity, SensorEntity):
+    """Define an AirVisual Pro sensor."""
+
+    _attr_has_entity_name = True
+
+    MEASUREMENTS_KEY_TO_VALUE = {
+        SENSOR_KIND_CO2: "co2",
+        SENSOR_KIND_HUMIDITY: "humidity",
+        SENSOR_KIND_PM_0_1: "pm0_1",
+        SENSOR_KIND_PM_1_0: "pm1_0",
+        SENSOR_KIND_PM_2_5: "pm2_5",
+        SENSOR_KIND_TEMPERATURE: "temperature_C",
+        SENSOR_KIND_VOC: "voc",
+    }
+
+    @property
+    def measurements(self) -> dict[str, Any]:
+        """Define measurements data."""
+        return self.coordinator.data["measurements"]
+
+    @property
+    def settings(self) -> dict[str, Any]:
+        """Define settings data."""
+        return self.coordinator.data["settings"]
+
+    @property
+    def status(self) -> dict[str, Any]:
+        """Define status data."""
+        return self.coordinator.data["status"]
+
+    @callback
+    def _async_update_from_latest_data(self) -> None:
+        """Update the entity from the latest data."""
+        if self.entity_description.key == SENSOR_KIND_AQI:
+            locale = async_get_aqi_locale(self.settings)
+            self._attr_native_value = self.measurements[locale]
+        elif self.entity_description.key == SENSOR_KIND_BATTERY_LEVEL:
```

Comment:
If we add a function to the description that looks up the correct coordinator value we could drop this if/else block and just implement the native value property.

Example:
https://github.com/home-assistant/core/blob/fc94569a0d807885141518e076871ab9b080ef90/homeassistant/components/wled/sensor.py#L34-L38

https://github.com/home-assistant/core/blob/fc94569a0d807885141518e076871ab9b080ef90/homeassistant/components/wled/sensor.py#L50-L60

https://github.com/home-assistant/core/blob/fc94569a0d807885141518e076871ab9b080ef90/homeassistant/components/wled/sensor.py#L163-L166

Reply:
https://github.com/home-assistant/core/pull/84898